### ISSUE TITLE
Segmentation fault in nvdla_runtime

### ISSUE THREAD

**Issue by Y159753** [nvdla/sw#29]
I have an issue when I test nvdla_runtime in the docker container, does anyone found a solution ?
Can it be because the format of my input image is not good ?
Does the image have to be 227x227 whatever the choosen neural network ? 

here are my terminal commands :  

 Welcome to Buildroot
buildroot login: root
Password:

 mount -t 9p -o trans=virtio r /mnt # mount pwd
 cd mnt
 insmod drm.ko
 insmod opendla.ko

[   80.649593] opendla: loading out-of-tree module taints kernel.
[   80.655147] reset engine done
[   80.659432] [drm] Initialized nvdla 0.0.0 20171017 for 10200000.nvdla on minor 0

ls

256x256dalmatian.jpg        drm.ko
256x256dalmatian.pgm        efi-virtio.rom
BDMA_L0_0_fbuf              init_dla.sh
Image                       libnvdla_runtime.so
LICENSE                     nvdla_runtime
aarch64_nvdla.lua           opendla.ko
aarch64_nvdla_dump_dts.lua  output.protobuf
default.nvdla               rootfs.ext4
 
./nvdla_runtime --loadable default.nvdla --image 256x256dalmatian.pgm

**./nvdla_runtime: error while loading shared libraries: libnvdla_runtime.so: cannot open shared object file: No such file or directory**

_(This is my solution to solve this error)_
 **echo $LD_LIBRARY_PATH
 LD_LIBRARY_PATH=/mnt
 export LD_LIBRARY_PATH**

 ./nvdla_runtime --loadable default.nvdla --image 256x256dalmatian.pgm
creating new runtime context...
Emulator starting
ppgminfo 1 256 256
pgm2dimg 1 256 256 1 256 65536 65536
[ 6978.529631] nvdla_runtime[1291]: unhandled level 3 translation fault (11) at 0xffff6f432000, esr 0x92000047, in libc-2.23.so[ffff85cd1000+12c000]
[ 6978.530786] CPU: 0 PID: 1291 Comm: nvdla_runtime Tainted: G           O    4.13.3 #1
[ 6978.531229] Hardware name: linux,dummy-virt (DT)
[ 6978.531618] task: ffff80003da80000 task.stack: ffff80003da50000
[ 6978.532238] PC is at 0xffff85d49494
[ 6978.534172] LR is at 0x409d8c
[ 6978.534437] pc : [<0000ffff85d49494>] lr : [<0000000000409d8c>] pstate: 20000000
[ 6978.534928] sp : 0000ffffe4d893a0
[ 6978.535209] x29: 0000ffffe4d893a0 x28: 0000000000000000
[ 6978.535619] x27: 0000000000000000 x26: 0000000000000000
[ 6978.535995] x25: 0000000000000000 x24: 0000000000000000
[ 6978.537816] x23: 0000000000000000 x22: 0000000000000000
[ 6978.538247] x21: 00000000004033a8 x20: 0000000000000000
[ 6978.538638] x19: 0000ffffe4d89f1f x18: 0000000000000000
[ 6978.539025] x17: 0000000000450710 x16: 0000ffff85d49380
[ 6978.539401] x15: 000000000000044e x14: 0000000000000000
[ 6978.539797] x13: 0000000000000000 x12: 0000000000000000
[ 6978.540201] x11: 0000000000005998 x10: 0000000000000000
[ 6978.542074] x9 : 0000000000000000 x8 : 0000000000000000
[ 6978.542449] x7 : 0000000000005930 x6 : 0000ffff6f431ff0
[ 6978.542796] x5 : 0000000000000040 x4 : 0000000000000000
[ 6978.543094] x3 : 0000ffff6f29f000 x2 : 000000000006cf80
[ 6978.543378] x1 : 0000ffff6f231040 x0 : 0000ffff6f29f000
Segmentation fault

**Comment by prasshantg** [comment]
Which network are you running here?

**Comment by Y159753** [comment]
The alex net from berkeley's examples from : https://github.com/BVLC/caffe/tree/master/models/bvlc_alexnet
it is supposed to take as input 256x256 pixels images.  

I just realised in the paper it is written that :   
"We did not pre-process the images
in any other way, _except for subtracting the mean activity over the training set from each pixel_. So
we trained our network on the (centered) raw RGB values of the pixels."
May it be the issue ?

**Comment by prasshantg** [comment]
It takes 227x227 input image

input_param { shape: { dim: 10 dim: 3 dim: 227 dim: 227 } }

**Comment by Y159753** [comment]
Thank you, you are right, the paper does not says that, I am totally begining with caffe. 
Apparently the "dim 10" comes from the fact the the input as to be 10 crops of a single image.
Each crop size of 227x227 I guess.  
https://stackoverflow.com/questions/32208193/how-to-modify-caffe-network-input-for-c-api

**Comment by prasshantg** [comment]
"dim 10" is batch size

**Comment by Y159753** [comment]
According to you 
Can I reduce the batch size in caffe just by replacing 10 by 1 and recompiling ? 
Or do I have to find a way to input {dim: 10 dim: 3 dim: 227 dim: 227} images ?

**Comment by prasshantg** [comment]
you can reduce batch size to 1 and recompile

**Comment by prasshantg** [comment]
Closing issue, please reopen if you are still observing issue
